File: app/src/main/java/acm/androidfitnesseexample/AndroidFitnesseExampleApplication.java
package acm.androidfitnesseexample;

import android.app.Application;
import android.os.AsyncTask;

import java.io.IOException;

import fitnesse.slim.JavaSlimFactory;
import fitnesse.slim.SlimService;

/**
 * Created in February.
 */
public class AndroidFitnesseExampleApplication extends Application {

    @Override
    public void onCreate() {
        new SlimClientTask().execute();
    }


    class SlimClientTask extends AsyncTask<String, Void, Void> {

        protected Void doInBackground(String... urls) {

            SlimService.Options options = SlimService.parseCommandLine(new String[]{"-v", "-d", "8099"});
            try {
                SlimService.startWithFactory(JavaSlimFactory.createJavaSlimFactory(options), options);
            } catch (IOException e) {
                e.printStackTrace();
            }
            return null;
        }
    }
}

File: app/src/main/java/acm/androidfitnesseexample/AndroidFitnesseExampleApplication.kt
package acm.androidfitnesseexample

import android.app.Application
import fitnesse.slim.JavaSlimFactory
import fitnesse.slim.SlimService
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import java.io.IOException

class AndroidFitnesseExampleApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        launchFitnesseServer()
    }

    private fun launchFitnesseServer() {
        val options = SlimService.parseCommandLine(arrayOf("-v", "-d", "8099"))
        GlobalScope.launch {
            try {
                SlimService.startWithFactory(JavaSlimFactory.createJavaSlimFactory(options), options)
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
    }
}